### PR TITLE
New style/core bit xxx

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/bit_notand.hpp [hunk 1]
```diff
@@ -8,10 +8,29 @@
...
 
 #include <eve/arch.hpp>
-#include <eve/detail/overload.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/traits/bit_value.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct bit_notand_t : tuple_callable<bit_notand_t, Options, numeric_option, pedantic_option, saturated_option>
```
Comment (jfalcou) [inline include/eve/module/core/regular/bit_notand.hpp:17]: why those options ?

File: include/eve/module/core/regular/bit_notor.hpp [hunk 2]
```diff
@@ -6,12 +6,30 @@
...
-
 #include <eve/arch.hpp>
-#include <eve/detail/overload.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/traits/bit_value.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct bit_notor_t : tuple_callable<bit_notor_t, Options, numeric_option, pedantic_option, saturated_option>
```
Comment (jfalcou) [inline include/eve/module/core/regular/bit_notor.hpp:16]: copy paste - why the options ??

File: include/eve/module/core/regular/bit_select.hpp [hunk 3]
```diff
@@ -61,15 +74,14 @@ namespace eve
 //!
 //!  @}
 //================================================================================================
-
+  inline constexpr auto bit_select = functor<bit_select_t>;
 namespace tag
```
Comment (jfalcou) [inline include/eve/module/core/regular/bit_select.hpp:78]: This old code is irrelevant, remove it

File: include/eve/module/core/regular/impl/simd/arm/neon/bit_andnot.hpp [hunk 4]
```diff
@@ -13,11 +13,12 @@
 
 namespace eve::detail
 {
-template<arithmetic_scalar_value T, typename N>
-EVE_FORCEINLINE wide<T, N>
-                bit_andnot_(EVE_SUPPORTS(neon128_),
-                            wide<T, N> const                &v0,
-                            wide<T, N> const                &v1) noexcept requires arm_abi<abi_t<T, N>>
+template<arithmetic_scalar_value T, typename N, callable_options O>
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/arm/neon/bit_andnot.hpp:16]: this arithmetic_scalar_value is unneeded. No concept in the xxx_ version

File: include/eve/module/core/regular/impl/simd/x86/bit_andnot.hpp [hunk 5]
```diff
@@ -10,59 +10,54 @@
...
-  if constexpr( c == category::float64x8 ) return _mm512_andnot_pd(v1, v0);
-  else if constexpr( c == category::float64x4 ) return _mm256_andnot_pd(v1, v0);
-  else if constexpr( c == category::float64x2 ) return _mm_andnot_pd(v1, v0);
-  else if constexpr( c == category::float32x16 ) return _mm512_andnot_ps(v1, v0);
-  else if constexpr( c == category::float32x8 ) return _mm256_andnot_ps(v1, v0);
-  else if constexpr( c == category::float32x4 ) return _mm_andnot_ps(v1, v0);
-  else if constexpr( i && std::same_as<abi_t<T, N>, x86_128_> ) return _mm_andnot_si128(v1, v0);
-  else if constexpr( i && std::same_as<abi_t<T, N>, x86_256_> )
+  // -----------------------------------------------------------------------------------------------
+  // 128 bits implementation
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/bit_andnot.hpp:21]: comment is not true anymore - change

File: include/eve/module/core/regular/impl/simd/x86/bit_notand.hpp [hunk 6]
```diff
@@ -17,70 +17,66 @@
...
-                bit_notand_(EVE_SUPPORTS(sse2_),
-                            wide<T, N> const                &v0,
-                            wide<T, N> const                &v1) noexcept requires std::same_as<abi_t<T, N>, x86_128_>
-{
-  if constexpr( std::is_same_v<T, float> ) return _mm_andnot_ps(v0, v1);
-  else if constexpr( std::is_same_v<T, double> ) return _mm_andnot_pd(v0, v1);
-  else if constexpr( std::is_integral_v<T> ) return _mm_andnot_si128(v0, v1);
-}
+  // -----------------------------------------------------------------------------------------------
+  // 128 bits implementation
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/bit_notand.hpp:21]: comment is out of date - change

**Review by jfalcou — CHANGES_REQUESTED** [review]
A bunch of details to clean up

File: include/eve/module/core/regular/bit_notand.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_notand.hpp:17]: suppressed

File: include/eve/module/core/regular/bit_notor.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_notor.hpp:16]: idem

File: include/eve/module/core/regular/bit_notand.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_notand.hpp:17]: idem

File: include/eve/module/core/regular/bit_select.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_select.hpp:78]: done

File: include/eve/module/core/regular/impl/simd/arm/neon/bit_andnot.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/arm/neon/bit_andnot.hpp:16]: done

File: include/eve/module/core/regular/impl/simd/x86/bit_andnot.hpp (hunk 5 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/bit_andnot.hpp:21]: done

File: include/eve/module/core/regular/impl/simd/x86/bit_notand.hpp (hunk 6 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/bit_notand.hpp:21]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**